src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3503,12 +3506,17 @@ CoPartitionedTables(Oid firstRelationId, Oid secondRelationId)
 		return true;
 	}
 
+	typeId = sortedFirstIntervalArray[0].valueTypeId;
+	comparisonFunction = GetFunctionInfo(typeId, BTREE_AM_OID, BTORDER_PROC);
```

Comment:
Though not directly related to this change: One note on finding the shard interval comparison function. In #416 we have which looks better since it covers hash partitioning as well:
```C
/*
 * ShardIntervalCompareFunction returns the appropriate compare function for the
 * partition column type. In case of hash-partitioning, it always returns the compare
 * function for integers.
 */
FmgrInfo *
ShardIntervalCompareFunction(Var *partitionColumn, char partitionMethod)
{
	FmgrInfo *compareFunction = NULL;

	if (partitionMethod == DISTRIBUTE_BY_HASH)
	{
		compareFunction = GetFunctionInfo(INT4OID, BTREE_AM_OID, BTORDER_PROC);
	}
	else if (partitionMethod == DISTRIBUTE_BY_RANGE)
	{
		compareFunction = GetFunctionInfo(partitionColumn->vartype,
										  BTREE_AM_OID, BTORDER_PROC);
	}
	else
	{
		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
						errmsg("unsupported partition method %d", partitionMethod)));
	}

	return compareFunction;
}
```

I think, for this PR we can copy&paste this function. Later on if we decide to merge it into master branch, we can use the original function.